### PR TITLE
fix memory leak in NSURLSession handling

### PR DESCRIPTION
**Pull request by olegnaumenko**

### CODE REVIEW

File: AFNetworking/AFURLSessionManager.m [hunk 1]
```diff
@@ -316,7 +316,8 @@ + (void)load {
             7) If the current class implementation of `resume` is not equal to the super class implementation of `resume` AND the current implementation of `resume` is not equal to the original implementation of `af_resume`, THEN swizzle the methods
             8) Set the current class to the super class, and repeat steps 3-8
          */
-        NSURLSessionDataTask *localDataTask = [[NSURLSession sessionWithConfiguration:nil] dataTaskWithURL:nil];
+        NSURLSession * session = [NSURLSession sessionWithConfiguration:nil];
```
Comment (alistra) [inline AFNetworking/AFURLSessionManager.m]: `+ (NSURLSession * nonnull)sessionWithConfiguration:(NSURLSessionConfiguration * nonnull)configuration`

This method actually requires an argument according to iOS 9 sdk nullability

File: AFNetworking/AFURLSessionManager.m [hunk 2]
```diff
@@ -316,7 +316,8 @@ + (void)load {
...
+        NSURLSessionDataTask *localDataTask = [session dataTaskWithURL:nil];
```
Comment (alistra) [inline AFNetworking/AFURLSessionManager.m]: `- (NSURLSessionDataTask * nullable)dataTaskWithURL:(NSURL * nonnull)url`

this too requires a parameter

**Comment by phoney** [comment]
This is a real bug that should be fixed. Apple's docs say

"The session object keeps a strong reference to the delegate until your app explicitly invalidates the session. If you do not invalidate the session, your app leaks memory."

alistra's comments are correct but the fix is trivial. Just add the default configuration when creating the session. I believe that @kcharwood wrote this code so maybe you want to fix this if @olegnaumenko isn't responding.

**Comment by kcharwood** [comment]
I'll take a look at this soon. Thanks @phoney

**Comment by kcharwood** [comment]
Rebased with some additional changes in 09f6cc1

**Comment by kcharwood** [comment]
Pushed to master in 09f6cc1

Pushed to 3.0.0 with b2c449c

Thanks everyone! :beers:

**Pull request closed without merging**